libqtile/backend/wayland/inputs.py

```diff
@@ -228,17 +233,33 @@ def finalize(self) -> None:
             if self.core.keyboards:
                 self.seat.set_keyboard(self.core.keyboards[-1].keyboard)
 
-    def set_keymap(self, layout: str | None, options: str | None, variant: str | None) -> None:
+    def set_keymap(
+        self,
+        layout: str | None,
+        options: str | None,
+        variant: str | None,
+        xkb_file: str | None = None,
+    ) -> None:
         """
         Set the keymap for this keyboard.
         """
-        if (layout, options, variant) in self._keymaps:
-            keymap = self._keymaps[(layout, options, variant)]
+        if (layout, options, variant, xkb_file) in self._keymaps:
+            keymap = self._keymaps[(layout, options, variant, xkb_file)]
+        elif xkb_file is not None:
+            xkbf = Path(xkb_file).expanduser()
+            if xkbf.is_file():
+                xkbf_name = xkbf.as_posix()
+            else:
+                raise ConfigError(f"kb_file {xkb_file} is not found.")
+
+            with open(xkbf_name, "r") as f:
+                keymap = self.xkb_context.keymap_new_from_file(f)
```

Comment:
move this to a function maybe?

Reply:
If we could use this in x11 then, yes, makes sense.